Segment/Internal/SEGFileStorage.m

```diff
@@ -134,7 +134,34 @@ + (NSURL *)applicationSupportDirectoryURL
 {
     NSArray *paths = NSSearchPathForDirectoriesInDomains(NSApplicationSupportDirectory, NSUserDomainMask, YES);
     NSString *storagePath = [paths firstObject];
+#if TARGET_OS_IPHONE
+    // will return ".../Application Support" because it's in a sandbox.
     return [NSURL fileURLWithPath:storagePath];
+#else
+    NSString *executableName = [[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleExecutable"];
+    // if we can't get an exec name, we're probably running as tests in a library of some kind, so make one up.
+    if (executableName == nil) {
+        executableName = @"segment-test";
```

Comment:
This does make me nervous in that I don't know all cases this can happen, but lets roll with it